example/matchers/consumer/tests/Service/MatchersTest.php

```diff
@@ -27,17 +27,26 @@ public function testGetMatchers(): void
             ->setMethod('GET')
             ->setPath($this->matcher->regex('/matchers', '^\/matchers$'))
             ->setQuery([
-                'pages' => $this->matcher->regex([1, 22], '\d+'), // arrayContains, eachKey, eachValue matchers are not working with query
-                'locales[]' => $this->matcher->regex(['en-US', 'en-AU'], '^[a-z]{2}-[A-Z]{2}$'), // Use `locales[]` instead of `locales` syntax if provider use PHP language
+                'pages' => $this->matcher->eachValue([1, 22], [
+                    $this->matcher->regex(null, '\d+')
+                ]),
+                // 'pages' => $this->matcher->regex([1, 22], '\d+'), // Alternative approach
+                'locales[]' => $this->matcher->eachValue(['en-US', 'en-AU'], [
+                    $this->matcher->regex(null, '^[a-z]{2}-[A-Z]{2}$'),
+                ]),  // Use `locales[]` instead of `locales` syntax if provider use PHP language
+                // 'locales[]' => $this->matcher->regex(['en-US', 'en-AU'], '^[a-z]{2}-[A-Z]{2}$'), // Alternative approach
```

Comment:
Do we need the commented alternative approaches?

Given it is in the examples directory, it might be worth keeping for pedagogical purposes, but I personally think it would be more readable as its own large stand-alone comment which might explain the pros and cons of both alternatives.

Reply:
I made some changes to make it more readable https://github.com/pact-foundation/pact-php/pull/554